### ISSUE TITLE
"Error decrypting preference #14" on fresh install

### ISSUE THREAD

**Issue by telyn** [JC5/firefly-iii#363]
On a fresh 4.1.3 install on Ubuntu 16.04 w/ MySQL 5.7 (though this also happens with MySQL 5.5), Firefly works fine until attempting to register an account, at which point I get the following error. Strangely I have noticed I get a similar error, but with preference #1 when using a 32-character APP_KEY instead of the new base64: style ones.

```
[2016-10-25 10:03:49] production.ERROR: Exception is: {"class":"FireflyIII\\Exceptions\\FireflyException","errorMessage":"Could not decrypt preference #14.","time":"Tue, 25 Oct 2016 10:03:49 +0000","file":"\/srv\/firefly\/app\/Models\/Preference.php","line":61,"code":0}
[2016-10-25 10:03:50] production.ERROR: FireflyIII\Exceptions\FireflyException: Could not decrypt preference #14. in /srv/firefly/app/Models/Preference.php:61
Stack trace:
#0 /srv/firefly/bootstrap/cache/compiled.php(11266): FireflyIII\Models\Preference->getDataAttribute('eyJpdiI6InlWV1d...')
#1 /srv/firefly/bootstrap/cache/compiled.php(11226): Illuminate\Database\Eloquent\Model->mutateAttribute('data', 'eyJpdiI6InlWV1d...')
#2 /srv/firefly/bootstrap/cache/compiled.php(11215): Illuminate\Database\Eloquent\Model->getAttributeValue('data')
#3 /srv/firefly/bootstrap/cache/compiled.php(11566): Illuminate\Database\Eloquent\Model->getAttribute('data')
#4 /srv/firefly/app/Http/Middleware/AuthenticateTwoFactor.php(56): Illuminate\Database\Eloquent\Model->__get('data')
#5 /srv/firefly/bootstrap/cache/compiled.php(9870): FireflyIII\Http\Middleware\AuthenticateTwoFactor->handle(Object(Illuminate\Http\Request), Object(Closure))
#6 /srv/firefly/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(33): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#7 /srv/firefly/app/Http/Middleware/Authenticate.php(53): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#8 /srv/firefly/bootstrap/cache/compiled.php(9870): FireflyIII\Http\Middleware\Authenticate->handle(Object(Illuminate\Http\Request), Object(Closure))
#9 /srv/firefly/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(33): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#10 /srv/firefly/bootstrap/cache/compiled.php(9978): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#11 /srv/firefly/bootstrap/cache/compiled.php(9870): Illuminate\Routing\Middleware\SubstituteBindings->handle(Object(Illuminate\Http\Request), Object(Closure))
#12 /srv/firefly/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(33): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#13 /srv/firefly/bootstrap/cache/compiled.php(3151): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#14 /srv/firefly/bootstrap/cache/compiled.php(9870): Illuminate\Foundation\Http\Middleware\VerifyCsrfToken->handle(Object(Illuminate\Http\Request), Object(Closure))
#15 /srv/firefly/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(33): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#16 /srv/firefly/bootstrap/cache/compiled.php(13467): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#17 /srv/firefly/bootstrap/cache/compiled.php(9870): Illuminate\View\Middleware\ShareErrorsFromSession->handle(Object(Illuminate\Http\Request), Object(Closure))
#18 /srv/firefly/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(33): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#19 /srv/firefly/bootstrap/cache/compiled.php(11907): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#20 /srv/firefly/bootstrap/cache/compiled.php(9870): Illuminate\Session\Middleware\StartSession->handle(Object(Illuminate\Http\Request), Object(Closure))
#21 /srv/firefly/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(33): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#22 /srv/firefly/bootstrap/cache/compiled.php(13213): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#23 /srv/firefly/bootstrap/cache/compiled.php(9870): Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse->handle(Object(Illuminate\Http\Request), Object(Closure))
#24 /srv/firefly/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(33): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#25 /srv/firefly/bootstrap/cache/compiled.php(13150): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#26 /srv/firefly/bootstrap/cache/compiled.php(9870): Illuminate\Cookie\Middleware\EncryptCookies->handle(Object(Illuminate\Http\Request), Object(Closure))
#27 /srv/firefly/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(33): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#28 /srv/firefly/bootstrap/cache/compiled.php(9855): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#29 /srv/firefly/bootstrap/cache/compiled.php(8148): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#30 /srv/firefly/bootstrap/cache/compiled.php(8139): Illuminate\Routing\Router->runRouteWithinStack(Object(Illuminate\Routing\Route), Object(Illuminate\Http\Request))
#31 /srv/firefly/bootstrap/cache/compiled.php(8130): Illuminate\Routing\Router->dispatchToRoute(Object(Illuminate\Http\Request))
#32 /srv/firefly/bootstrap/cache/compiled.php(2472): Illuminate\Routing\Router->dispatch(Object(Illuminate\Http\Request))
#33 /srv/firefly/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\Foundation\Http\Kernel->Illuminate\Foundation\Http\{closure}(Object(Illuminate\Http\Request))
#34 /srv/firefly/bootstrap/cache/compiled.php(3213): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#35 /srv/firefly/bootstrap/cache/compiled.php(9870): Illuminate\Foundation\Http\Middleware\CheckForMaintenanceMode->handle(Object(Illuminate\Http\Request), Object(Closure))
#36 /srv/firefly/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(33): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#37 /srv/firefly/bootstrap/cache/compiled.php(9855): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#38 /srv/firefly/bootstrap/cache/compiled.php(2416): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#39 /srv/firefly/bootstrap/cache/compiled.php(2400): Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter(Object(Illuminate\Http\Request))
#40 /srv/firefly/public/index.php(62): Illuminate\Foundation\Http\Kernel->handle(Object(Illuminate\Http\Request))
#41 {main}
```

**Comment by JC5** [comment]
Preferences are cached, and may be stored in cache using the old key. Try
"php artisan cache:clear".

**Comment by telyn** [comment]
Yep, that fixed it. Thanks / sorry! Laravel is not my home framework...

**Comment by telyn** [comment]
Alright, I managed to get upgraded to from 3.7.2.3 to 4.1.3 and everything works fine. I used MySQL Workbench to create an alteration script which almost entirely worked automatically (I just had to `UPDATE preferences SET data = encrypted_data` before running the generated script)

Thanks for all your help, and for making such a great piece of software, and implementing that feature I asked for! And now I am going to go add lots of notes to my piggy banks.
